srcStatic/WindowsModule.cpp

```diff
@@ -8,19 +8,20 @@
 #undef _UNICODE
 #include <windows.h>
 #include <tlhelp32.h> // CreateToolhelp32Snapshot, Module32First, Module32Next
+#include "WindowsUniqueHandle.h"
+
 
 std::string FindModuleName(HANDLE hModuleSnap, const void* address);
 bool containsAddress(MODULEENTRY32 const& moduleEntry, const void* address);
 
 
 std::string FindModuleName(const void* address) {
 	// Get all modules for current process (processId can be 0)
-	auto hModuleSnap = CreateToolhelp32Snapshot(TH32CS_SNAPMODULE, 0);
-	if (hModuleSnap == INVALID_HANDLE_VALUE) {
+	UniqueHandleOrInvalid hModuleSnap{CreateToolhelp32Snapshot(TH32CS_SNAPMODULE, 0)};
+	if (!hModuleSnap) {
 		return std::string("<Unable to create module snapshot>");
 	}
-	auto moduleName = FindModuleName(hModuleSnap, address);
-	CloseHandle(hModuleSnap);
+	auto moduleName = FindModuleName(hModuleSnap.get(), address);
```

Comment:
Might make sense to remove this variable and just directly return the result of FindModuleName.

```C++
return FindModuleName(hModuleSnap.get(), address);
```

Reply:
Good point. I wasn't even looking at the code at that high of a level.